Sources/Component/CollapsibleView/CollapsibleView.swift

```diff
@@ -0,0 +1,132 @@
+        actionButton.addTarget(self, action: #selector(updateBackgroundColor), for: .allTouchEvents)
+        actionButton.addTarget(self, action: #selector(lateUpdateBackgroundColor), for: .allTouchEvents)
+        return actionButton
+    }()
+
+    private lazy var stackView: UIStackView = {
+        let stackView: UIStackView = .init()
+        stackView.alignment = .fill
+        stackView.axis = .vertical
+        stackView.distribution = .fillProportionally
+        return stackView
+    }()
+
+    public override func viewDidLoad() {
+        super.viewDidLoad()
+
+        headerViewContainer.backgroundColor = .white
+        headerViewContainer.translatesAutoresizingMaskIntoConstraints = false
+        addSubview(headerViewContainer)
+        headerViewContainer.topAnchor.constraint(equalTo: topAnchor).isActive = true
+        headerViewContainer.leadingAnchor.constraint(equalTo: leadingAnchor).isActive = true
+        headerViewContainer.trailingAnchor.constraint(equalTo: trailingAnchor).isActive = true
+
+        actionButton.translatesAutoresizingMaskIntoConstraints = false
+        addSubview(actionButton)
+        actionButton.topAnchor.constraint(equalTo: headerViewContainer.topAnchor).isActive = true
+        actionButton.leadingAnchor.constraint(equalTo: headerViewContainer.leadingAnchor).isActive = true
+        actionButton.trailingAnchor.constraint(equalTo: headerViewContainer.trailingAnchor).isActive = true
+        actionButton.bottomAnchor.constraint(equalTo: headerViewContainer.bottomAnchor).isActive = true
+
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        addSubview(stackView)
+        stackView.topAnchor.constraint(equalTo: headerViewContainer.bottomAnchor).isActive = true
+        stackView.leadingAnchor.constraint(equalTo: leadingAnchor).isActive = true
+        stackView.trailingAnchor.constraint(equalTo: trailingAnchor).isActive = true
+        stackView.bottomAnchor.constraint(equalTo: bottomAnchor).isActive = true
+    }
+
+    public override func didChangeModel() {
+        super.didChangeModel()
+
+        model.headerView.translatesAutoresizingMaskIntoConstraints = false
+        headerViewContainer.addSubview(model.headerView)
+        model.headerView.topAnchor.constraint(equalTo: headerViewContainer.topAnchor).isActive = true
+        model.headerView.leadingAnchor.constraint(equalTo: headerViewContainer.leadingAnchor).isActive = true
+        model.headerView.trailingAnchor.constraint(equalTo: headerViewContainer.trailingAnchor).isActive = true
+        model.headerView.bottomAnchor.constraint(equalTo: headerViewContainer.bottomAnchor).isActive = true
+
+        if (stackView.arrangedSubviews.count != model.items.count) {
+            stackView.arrangedSubviews.forEach { $0.removeFromSuperview() }
+
+            model.items.forEach { view in
+                view.isHidden = !model.isCollapsed
+                stackView.addArrangedSubview(view)
+            }
```

Comment:
Maybe we could extract the content of this if condition into a function to reduce complexity and increase readability.

Reply:
Personally I would leave it as is. IMHO it is not that complex and adding more functions would increase the complexity otherwise. Maybe we should discuss this again or ask others to review it.